content/3_Silver/Two_Pointers.mdx

```diff
@@ -44,15 +45,145 @@ of an interval. It can also be used to track two values in an array as shown in
+We can then continually increment $\texttt{left}$ by 1. After incrementing the left index, the time used decreases, hence the right pointer never has to move back (left). Thus:
+
+**Since both pointers will move at most $N$ times, the overall time complexity is $\mathcal{O}(N)$.**
+
+As an example, consider the first case in the sample inputs:
+<center>
+<table>
+	<tbody>
+		<tr>
+			<td>Left Pointer</td>
+			<th className="bg-yellow-100 dark:bg-yellow-800"></th>
+			<th></th>
+			<th></th>
+			<th></th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>3</td>
+			<td>1</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>Right Pointer</td>
+			<th className="bg-yellow-100 dark:bg-yellow-800"></th>
+			<th></th>
+			<th></th>
+			<th></th>
+		</tr>
+	</tbody>
+</table>
+</center>
+We can move the right pointer to index $1$:
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Left Pointer</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800"></th>
+			<th></th>
+			<th></th>
+			<th></th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>3</td>
+			<td>1</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<th>Right Pointer</th>
+			<th></th>
+			<th className="bg-yellow-100 dark:bg-yellow-800"></th>
+			<th></th>
+			<th></th>
+		</tr>
+	</tbody>
+</table>
+</center>
+The sum of the values in this range is 4, and there are 2 values. So, the current maximum is 2. By incrementing the left pointer by 1, we can subtract 3 from the sum of values to get 1. The array then looks like:
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Left Pointer</th>
+			<th></th>
+			<th className="bg-yellow-100 dark:bg-yellow-800"></th>
+			<th></th>
+			<th></th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>3</td>
+			<td>1</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<th>Right Pointer</th>
+			<th></th>
+			<th className="bg-yellow-100 dark:bg-yellow-800"></th>
+			<th></th>
+			<th></th>
+		</tr>
+	</tbody>
+</table>
+</center>
+Now, we can move the right pointer to the end. This makes the sum of values $4$, since $1$ (previous cur val) $+2+1=4$, and has a length of 3. So, the maximum is now $3$.
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Left Pointer</th>
+			<th></th>
+			<th className="bg-yellow-100 dark:bg-yellow-800"></th>
+			<th></th>
+			<th></th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>3</td>
+			<td>1</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<th>Right Pointer</th>
+			<th></th>
+			<th></th>
+			<th></th>
+			<th className="bg-yellow-100 dark:bg-yellow-800"></th>
+		</tr>
+	</tbody>
+</table>
+</center>
+At this point we are done, since the right pointer has reached the end. So, the answer is $3$.
```

Comment:
You just used "so" as the transition word two sentences ago, so it might be better to switch it up 
```suggestion
Since the right pointer has reached the end of the array, we are done at this point. This leaves us with $3$ as our answer.
```